src/main/java/org/commcare/cases/util/QueryUtils.java

```diff
@@ -0,0 +1,36 @@
+package org.commcare.cases.util;
+
+import org.commcare.cases.query.IndexedValueLookup;
+import org.commcare.cases.query.QueryContext;
+import org.commcare.cases.query.QuerySensitive;
+
+import java.util.Vector;
+
+/**
+ * Created by ctsims on 1/25/2017.
+ */
+
+public class QueryUtils {
+    public static org.commcare.cases.query.IndexedValueLookup getFirstKeyIndexedValue(Vector<org.commcare.cases.query.PredicateProfile> profiles) {
+        if(profiles.elementAt(0) instanceof org.commcare.cases.query.IndexedValueLookup) {
+            return (IndexedValueLookup)profiles.elementAt(0);
+        }
+        return null;
+    }
+
+    public static Vector<Integer> wrapSingleResult(Integer result) {
+        Vector<Integer> results = new Vector<>();
+        results.add(result);
+        return results;
+    }
+
+    public static void poke(Object o, QueryContext context) {
```

Comment:
Is this wrapper method necessary? Seems like `notifySensitive` could contain all of this logic. `poke()` is also a very ambiguous method name.